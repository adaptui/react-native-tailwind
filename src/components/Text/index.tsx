import { Text as RNText, TextProps as RNTextProps } from 'react-native';
import { createComponent } from '../../utils/createComponent';
import { createElement } from '../../utils/createElement';

export type LibraryTextProps = {};

export type TextProps = RNTextProps & LibraryTextProps;

export const Text = createComponent<TextProps>(
  (props) => {
    return createElement({
      componentType: RNText,
      props: props,
    });
  },
  { shouldMemo: true }
);
