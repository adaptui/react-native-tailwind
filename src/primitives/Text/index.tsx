import { Text as RNText, TextProps as RNTextProps } from 'react-native';
import type { Dict } from 'src/utils/types';
import { createComponent } from '../../utils/createComponent';

export type LibraryTextProps = Dict<unknown>;

export type TextProps = RNTextProps & LibraryTextProps;

export const Text = createComponent<TextProps>(RNText, { shouldMemo: true });
