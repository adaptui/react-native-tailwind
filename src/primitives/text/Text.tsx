import { Text as RNText } from "react-native";

import {
  As,
  createComponent,
  createElement,
  createHook,
  Options,
  Props,
} from "../../utils/system";

export const useText = createHook<TextOptions>(props => {
  return props;
});

export const Text = createComponent<TextOptions>(props => {
  const htmlProps = useText(props);
  return createElement(RNText, htmlProps);
});

export type TextOptions<T extends As = typeof RNText> = Options<T>;

export type TextProps<T extends As = typeof RNText> = Props<TextOptions<T>>;
