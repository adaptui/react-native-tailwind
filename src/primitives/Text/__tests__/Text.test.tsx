import { render } from '@testing-library/react-native';
import React from 'react';
import { Text } from '../../Text';

describe('basic', () => {
  it('should render correctly', () => {
    const component = render(<Text>Hello, World</Text>);
    expect(component).toMatchSnapshot();
  });

  it('should assign a ref', () => {
    const ref = React.createRef();
    render(<Text ref={ref}>Hello world</Text>);
    expect(ref.current).toMatchSnapshot();
  });

  it('should render correctly with a style prop', () => {
    const component = render(
      <Text
        style={{
          color: 'tomato',
          padding: 4,
          margin: 10,
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const component = render(
      <Text
        style={{
          color: 'tomato',
          padding: 4,
          margin: 10,
        }}
      >
        <Text
          style={{
            color: 'white',
            textAlign: 'center',
            fontSize: 18,
          }}
        >
          This is a Text with Text Inside
        </Text>
        This is outside Text
      </Text>
    );
    expect(component).toMatchSnapshot();
  });
});
