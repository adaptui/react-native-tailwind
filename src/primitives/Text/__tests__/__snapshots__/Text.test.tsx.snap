// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should assign a ref 1`] = `null`;

exports[`basic should render children correctly 1`] = `
<Text
  elementRef={null}
  style={
    Object {
      "color": "tomato",
      "margin": 10,
      "padding": 4,
    }
  }
>
  <Text
    elementRef={null}
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    This is a Text with Text Inside
  </Text>
  This is outside Text
</Text>
`;

exports[`basic should render correctly 1`] = `
<Text
  elementRef={null}
>
  Hello, World
</Text>
`;

exports[`basic should render correctly with a style prop 1`] = `
<Text
  elementRef={null}
  style={
    Object {
      "color": "tomato",
      "margin": 10,
      "padding": 4,
    }
  }
/>
`;
